### DISCUSSION TITLE
How can I use python extesions?

### DISCUSSION THREAD

**Discussion by NailGit** [ahrm/sioyek#820] [General]
I found Python extensions https://github.com/ahrm/sioyek-python-extensions , and installed them by  `python3 -m pip install sioyek`. Could you explain a little bit more about how to use them? I'm interested in a dual-pages extension. I compiled Sioyek and copied binary in /usr/local/bin. I put in the pref_user file `new_command _dual_panelify python -m sioyek.dual_panelify "%{sioyek_path}" "%{file_path}" "%{command_text}"`. The command `_dual_panelify ` appears but it does not work.

**Comment by ahrm** [comment]
You can run the command 
```
python -m sioyek.dual_panelify "%{sioyek_path}" "%{file_path}" "%{command_text}"
```
(after replacing %{} variables with the correct values). In the command line and see if there are any errors.

**Reply by NailGit** [reply]
Yes, it works, but started 3 times. I closed the window and it opened 2 times more.  But how make a dual regime inside the opened Sioyek? Without a command line.

**Reply by ahrm** [reply]
If the command works in the terminal it should also work within sioyek. Did you modify the command in any way (except replacing the paths) to make it work?

**Reply by NailGit** [reply]
I used command `python3 -m sioyek.dual_panelify "/path/to/sioiyek" "/path/to/file" "25 -40"` . But I do not understand - each time I must put the specific file to this command in sloiyek? I respected that I could use a dual regime and switch off it for any file without changing the command.

**Reply by ahrm** [reply]
No, sioyek would automatically replace the "%{}" variables with the correct paths.

**Reply by NailGit** [reply]
It starts to work with the command `new_command _dual_panelify python3 -m sioyek.dual_panelify "%{sioyek_path}" "%{file_path}" "25 -40"`. It works strangely - the dual regime is opened in a new window of Sioyek and a new file file_dual_panel.pdf is created. After closing the dual regime window, this file is not deleted. It is not convenient. 

You wrote that you have a plan to implement a dual regime in Sioyek. It is better to make a dual regime in the same window as in the majority of viewers without new file creation.

**Reply by ahrm** [reply]
Sioyek doesn't support dual panel right now. It is explained in the description of the extension that a dual panel version of current file is created.
